packages/pg/lib/client.js

```diff
@@ -297,6 +727,13 @@ class Client extends EventEmitter {
   _handleErrorWhileConnecting(err) {
     if (this._connectionError) {
       // TODO(bmc): this is swallowing errors - we shouldn't do this
+      if (this.connectionParameters.load_balance) {
```

Comment:
It is required here because when one connection fails which marks `this._connectionError` as true and smart-driver tries to connect with other nodes and that node also fails to connect then it will again throw error and this `_handleErrorWhileConnecting ` is called and it will not emit the error as previously it is just returning from here if  `this._connectionError` is true . So, that is the case why we are explicitly calling callback for `loadBalance=true` here to try with other nodes now or it can be handled in `nowConnect()` [here](https://github.com/yugabyte/node-postgres/blob/b90c9e5c87508696e19b4e6c43e34632c0f3895e/packages/pg/lib/client.js#L474), where we are not calling callback to throw error and just calling `this.connect(callback)` to try connecting with other nodes.